### DISCUSSION TITLE
setValue can not update array data when its has defaultvalue.

### DISCUSSION THREAD

**Discussion by tonyYo00** [react-hook-form/react-hook-form#5678] [Q&A]
**Describe the bug**
setValue can not update array data when its has defaultvalue. its also happen in useFieldArray  ,when remove row, [object] will lost

**To Reproduce**

**Codesandbox link (Required)**
Include a codesandbox will help us to investigate the issue quicker.

 https://codesandbox.io/s/controlled-field-array-forked-xjfl9?file=/src/App.tsx

just click submit and see the log.

- Browser [browser,]
- Version [7.88]

**Comment by tonyYo00** [comment]
![Peek 2021-06-18 17-45](https://user-images.githubusercontent.com/79240024/122542320-25399a00-d05d-11eb-9b40-15dfe1e485c5.gif)

**Comment by bluebill1049** [comment]
we don't support `undefined` for `defaultValues`.

**Reply by tonyYo00** [reply]
ok. i update the defaultValue, but it  still has error .https://codesandbox.io/s/controlled-field-array-forked-xjfl9?file=/src/App.tsx

**Reply by bluebill1049** [reply]
I will take a look at this tmr.

**Comment by bluebill1049** [comment]
what's the steps to reproduce the issue in your CSB?

**Reply by tonyYo00** [reply]
whats mean CSB?

**Reply by bluebill1049** [reply]
codesandbox?

**Comment by bluebill1049** [comment]
## problem

https://react-hook-form.com/api/useform
<img width="904" alt="Screen Shot 2021-06-18 at 10 03 31 pm" src="https://user-images.githubusercontent.com/10513364/122558107-05b46880-d081-11eb-8eed-ddf825469f21.png">

```
const { handleSubmit, setValue, control, register } = useForm<FormValues>({
  defaultValues: {
    test: [{ test: new Date() }]
  }
});
```

because you don't really have a `test` input, and react hook form will register the above defaultValues into 

`test.0.test`

and when you try to update the input with `setValue('test')` the update is getting lost, so to work around this issue. 

## solutions

have a custom register or use controller

`register('test')`

or https://react-hook-form.com/api/usefieldarray

<img width="944" alt="Screen Shot 2021-06-18 at 10 06 00 pm" src="https://user-images.githubusercontent.com/10513364/122558396-5f1c9780-d081-11eb-85e3-60c62a779e92.png">
